### PR TITLE
update ACF example

### PR DESCRIPTION
**Pull request by macbookandrew**
As currently documented with version `1.2.3`, the command `composer require "advanced-custom-fields/advanced-custom-fields-pro:*"` fails.

This brings the version up to something close to current so it doesn’t immediately fail and discourage use of this package.

An alternative might be to use a placeholder instead of `1.2.3` so it’s a bit more obvious that the user needs to set it to a current version.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Oh nooooooooooooooooooooooooooo, @macbookandrew!

kép

That code snippet is in the **Examples** section of the README.

**Comment by szepeviktor** [comment]
I know - I know...
_Copypaste_ guys are in trouble 😢

**Comment by macbookandrew** [comment]
What do you think of this to make it more immediately obvious?

```diff
-    "version": "1.2.3",
+    "version": "{enter a specific version number}",
```

**Comment by szepeviktor** [comment]
I think copypasting before acquiring knowledge is something I no not dare to comment!

**Comment by szepeviktor** [comment]
On one hand supporting copypasters makes web development an even worse profession.
On the other hand giving a valid version makes the package more popular.

# 💣

**Comment by macbookandrew** [comment]
I sympathize with the dilemma. Maybe something like this would be better to encourage people to read more closely?

```diff
-    "version": "1.2.3",
+    "version": "1.2.3", # Note: you must replace this version number
```

The example _looks_ like a valid copy/pasteable snippet and when it fails throwing a `Composer\Download\TransportException`, it’s not apparent that it’s the user’s fault for blindly copy/pasting, possibly leading users to falsely assume this package is at fault.

Whatever you think; it’s your package 🙂 

Thanks for your work on it!

**Comment by szepeviktor** [comment]
> a valid copy/pasteable snippet

Actually the way you upgrade ACF is to edit `composer.json`, so when installing ACF the first time you should set the most recent version from ACF's changelog.

> it’s your package 🙂

It's @ffraenz's.

**Comment by macbookandrew** [comment]
Well then…it’s up to @ffraenz and I thank him for his work on it 🙂 

I’m just hoping to help those not as experienced as you or I so they _can_ learn something rather than assuming this is broken. Have a nice day 👋🏻

**Comment by szepeviktor** [comment]
> to help those not as experienced

Please consider adding a new section instead.

## Upgrade

This is how you upgrade a package installed by `private-composer-installer` ...

1. manual/scripted watch of new releases
2. update version constrain as needed
3. update version in package definition

**Comment by macbookandrew** [comment]
Yeah, that would be helpful too

**Comment by ffraenz** [comment]
Hi @macbookandrew, thank you for opening an issue on this. I see why the examples section might cause confusion or even frustration. I don't want to mention a specific ACF version in the README as we would need to keep it updated afterwards. Introducing a new placeholder format like `{placeholder}` could be confusing, too, as people could mistake it with the placeholders used in the dist URL. As comments are not allowed in JSON I would suggest to replace `1.2.3` by something like `REPLACE_BY_LATEST_ACF_VERSION`. For consistency let us also change the version in the 'Arbitrary private package' section.

@szepeviktor Maybe we could add an FAQ section to the README to include questions like 'How to update a package?'. We could also discuss this in a separate issue or pull request.

**Comment by macbookandrew** [comment]
> I don't want to mention a specific ACF version in the README as we would need to keep it updated afterwards.

That makes total sense.

I updated the PR with your requested changes. Thanks!

File: README.md [hunk 1]
```diff
@@ -21,14 +21,14 @@ This is a [Composer](https://getcomposer.org/) plugin offering a way to referenc
...
-Add the desired private package to the `repositories` field inside `composer.json`. Find more about Composer repositories in the [Composer documentation](https://getcomposer.org/doc/05-repositories.md#repositories). Replace the version as well as other sensitive tokens by `{%VARIABLE}` placeholders.
+Add the desired private package to the `repositories` field inside `composer.json`. Find more about Composer repositories in the [Composer documentation](https://getcomposer.org/doc/05-repositories.md#repositories). Specify the exact version to install, and use `{%VARIABLE}` placeholders to specify any sensitive tokens in your `.env` file.
 
 ```json
 {
   "type": "package",
   "package": {
     "name": "package-name/package-name",
-    "version": "1.0.0",
+    "version": "REPLACE_WITH_LATEST_PLUGIN_VERSION",
```
Comment (ffraenz) [inline README.md]: Let's change this to `REPLACE_WITH_LATEST_PACKAGE_VERSION` as it may not be a plugin.

**Review by ffraenz — CHANGES_REQUESTED** [review]
Just a small adjustment and we're good to go on this one 👍

File: README.md (hunk 1 above)
Comment (szepeviktor) [reply README.md]: Do we really need underscores? Resembles the DOS era :)

**Review by szepeviktor — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ffraenz) [reply README.md]: In my opinion it should be very clear that one single value is expected. That's why I'd go with the DOS style uppercase placeholder with underscores. Also I like the retro vibes ;)

Comment (szepeviktor) [reply README.md]: I see. This smells like sub-kindergarten.

kép

Comment (macbookandrew) [reply README.md]: > Let's change this to REPLACE_WITH_LATEST_PACKAGE_VERSION as it may not be a plugin.

Of course!

**Review by ffraenz — APPROVED** [review]

**Comment by ffraenz** [comment]
@macbookandrew Thanks again!

I will have a look at the failing tests in a later session. As we only updated README, it is not related to this PR.

**Comment by macbookandrew** [comment]
@ffraenz  any chance you could add `hacktoberfest-accepted` label to this?

**Comment by ffraenz** [comment]
Yeah, sure :)

**Pull request merged**